example/example/spiders/books_06.py

```diff
@@ -12,34 +12,34 @@
 import scrapy
 import attr
 
-from scrapy_po import WebPage, ItemPage
+from scrapy_po import WebPage, ItemWebPage, Injectable
 
 
-class ListingsPiece(WebPage):
+class ListingsExtractor(WebPage):
     def urls(self):
         return self.css('.image_container a::attr(href)').getall()
 
 
-class PaginationPiece(WebPage):
+class PaginationExtractor(WebPage):
     def urls(self):
         return self.css('.pager a::attr(href)').getall()
 
 
-class BreadcrumbsPiece(WebPage):
+class BreadcrumbsExtractor(WebPage):
     def urls(self):
         return self.css('.breadcrumb a::attr(href)').getall()
 
 
 @attr.s(auto_attribs=True)
-class ListingsPage(WebPage):
-    book_list: ListingsPiece
-    pagination: PaginationPiece
+class ListingsPage(Injectable):
+    book_list: ListingsExtractor
+    pagination: PaginationExtractor
 
 
 @attr.s(auto_attribs=True)
-class BookPage(WebPage, ItemPage):
-    recently_viewed: ListingsPiece
-    breadcrumbs: BreadcrumbsPiece
+class BookPage(ItemWebPage):
```

Comment:
`ItemWebPage` was introduced because `BookPage` must be a `WebPage` but also it should be an `ItemPage`. An alternative is to declare it here:
```py
class BookPage(WebPage, ItemPage):
    ....
```

Another alternative would have been to declare `ItemPage` as a `WebPage`:
```py
@attr.s(auto_attribs=True)
class ItemPage(WebPage):
    @abc.abstractmethod
    def to_item(self):
        pass
```
But then it wouldn't be allowed `ItemPages` that don't require `ResponseData`, and this is required for autoextract.